### ISSUE TITLE
bucket object delete issue

### ISSUE THREAD

**Issue by hsharma-lsquared** [minio/minio#4173]
All files to be deleted are csv files,

## Expected Behavior
Bucket object should be deleted.

## Current Behavior
Bucket object not deleted, remain as it is after deletion.

## Steps to Reproduce (for bugs)
delete object using following json and hit POST request on http://192.168.1.55:9000/minio/webrpc:
{"id":1,"jsonrpc":"2.0","params":{"bucketName":"myBucket","objectName":"folder1/folder2/filename"},"method":"Web.RemoveObject"}

with header:
authorization: Bearer TOKEN
content-type: application/json

## Your Environment
* Version used: RELEASE.2017-02-15T22-55-24Z
* Nginx
* Ubuntu 16.04

**Comment by harshavardhana** [comment]
Can you update to latest release and test again @hsharma-lsquared ?

**Comment by hsharma-lsquared** [comment]
My Minio is on production, Can't update there,

Before some time deletion was working correctly, but on new module same code couldn't delete object, for new module i created new bucket manually, using Minio Browser,

**Comment by harshavardhana** [comment]
> Before some time deletion was working correctly, but on new module same code couldn't delete object, for new module i created new bucket manually, using Minio Browser,

There was an odd issue which we fixed in a release after your current Minio version, looks like following is the possible fix. For which we have already made a release. 

```
commit 0bae3330e8c8910c69e8229a3f0268e6127b6982
Author: Krishna Srinivas <krishna@minio.io>
Date:   Fri Mar 3 17:07:17 2017 -0800

    browser: Send correct arguments for RemoveObjects web handler. (#3848)
    
    Fixes #3839
```

**Comment by hsharma-lsquared** [comment]
On development environment i tried using latest minio, it is so bad that in new version "bucketName" changed to "bucketname", also "objectName" changed to "objects" for multiple deletion, also i investigate prior releases, in them also these type of minor but for me major changes,

In new release why you remove old features, What happened to those users who are using minio at production, and you made new releases every month, sometimes twice in a month,

**Comment by harshavardhana** [comment]
> On development environment i tried using latest minio, it is so bad that in new version "bucketName" changed to "bucketname", also "objectName" changed to "objects" for multiple deletion, also i investigate prior releases, in them also these type of minor but for me major changes,
> 
> In new release why you remove old features, What happened to those users who are using minio at production, and you made new releases every month, sometimes twice in a month,

Are you using Web API directly? without using browser?

**Comment by hsharma-lsquared** [comment]
Version: 2017-02-16T01:47:30Z
Release-Tag: RELEASE.2017-02-16T01-47-30Z
Commit-ID: 3d98311d9f4ceb78dba996dcdc0751253241e697

this is full detail for you,

**Comment by hsharma-lsquared** [comment]
Yes, using API directly for my website,

**Comment by hsharma-lsquared** [comment]
Also see this, after a lot of efforts i found Minio version,
http://stackoverflow.com/q/43540631/5026957

**Comment by harshavardhana** [comment]
> Yes, using API directly for my website,

We never really published Web Handlers API neither it is documented anywhere as a recommended mechanism to access Minio Server. Neither it is versioned API so there is no set guarantee that it might not change in the future. Published way to access Minio is using AWS S3 API which is also going to be backward compatible.

Web Handlers are only meant to be used by Minio Browser and only serve Minio Browsers requirement.  We might even remove the API and just use S3 API if that makes it easier in future. 

So expect changes in web-handlers, if you rely on it. It is a design mistake - you should rely on AWS S3 API instead. Please re-do your application.

**Comment by hsharma-lsquared** [comment]
ok thank you for support.

**Comment by harshavardhana** [comment]
Fixed and release has been made closing.

```
commit 0bae3330e8c8910c69e8229a3f0268e6127b6982
Author: Krishna Srinivas <krishna@minio.io>
Date:   Fri Mar 3 17:07:17 2017 -0800

    browser: Send correct arguments for RemoveObjects web handler. (#3848)
    
    Fixes #3839
```
